kolibri/plugins/user_profile/assets/src/machines/changeFacilityMachine.js

```diff
@@ -209,7 +209,7 @@ export const changeFacilityMachine = createMachine({
       meta: { route: 'CONFIRM_MERGE', path: '/change_facility' },
       on: {
         CONTINUE: 'syncChangeFacility',
-        BACK: 'changeFacility',
+        BACK: 'confirmAccountDetails',
```

Comment:
We don't always want to navigate to 'confirmAccountDetails', in some cases we need to navigate to the choose a new super admin as illustrated on designs:
![Screenshot from 2022-09-06 15-24-04](https://user-images.githubusercontent.com/13509191/188646676-0a86b2ac-2c86-4c2d-bc9b-6d53185611fb.png)

We already have something similar implemented [here](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/user_profile/assets/src/machines/changeFacilityMachine.js#L223-L232)

I think that this should be resolved before merging.

Reply:
`confirmAccountDetails` takes you to `isAdmin` that determines if the Choose admin screen must be used:  
```
    isAdmin: {
      always: [
        {
          cond: context => context.role === 'superuser',
          target: 'chooseAdmin',
        },
        {
          target: 'checkIsMerging',
        },
      ],
    },

....

confirmAccountDetails: {
      meta: { route: 'CONFIRM_DETAILS', path: '/change_facility' },
      on: {
        CONTINUE: 'isAdmin',
        BACK: 'mergeAccounts',
      },
    },
```

I see the case where you're already in `chooseAdmin` , then go to `confirmMerge` and then back to a screen previous to `chooseAdmin` but I wonder if it's worth to add more conditions just for this case that will be rare and will take you to the same path anyway. If you think it's worth I'll add it but with the current complexity of the machine I'll rather leave it as it is because I don't see it as a problem.

Reply:
> with the current complexity of the machine I'll rather leave it as it is because I don't see it as a problem

Yes, you're right that those conditions make things pretty complex though I don't think that ignoring back navigation in some cases because of that complexity is a good idea and if so, I'd suggest discussing this with Jessica because on the designs it seems that "Back" button is meant to take you to the correct previous step in all flows? 

I can understand your concerns about adding more conditions, though, as since this problem is not related only to this particular PR, I'd suggest we leave it this way here but make sure to resolve that complexity on higher-level as a follow-up issue before we accumulate tech debt. For similar reasons, I was already pondering a bit about whether it'd make sense to introduce using nested states. Even from designs, it seems to be pretty intuitive to go in that direction. I imagine we could have three root states corresponding to the three main branches of designs (default path, username exists, merge accounts). Those would then contain sub-states representing steps of that particular path. It shouldn't be anything difficult to implement, we could get rid of the majority of conditions that make the machine complicated and I believe it'd make it much better readable. What do you think of this direction?

Reply:
Honestly, I don't like nested states too much. I only see them useful when there really 2 states to consider at the same time (as one element that's visible or not, and, if, visible, it's in one color or another).
I don't see they add too much value or reduce code or simplicity in other cases.

Reply:
so, do you want me to change it or do you want to open an issue to change it in the future? I just want to finish this simple task, if you prefer me to add a new state I'll rather do it than continue seeing days pass without a solution.

Reply:
Either of those will do

Reply:
Thanks, I've implemented it to avoid needing to work on this again in the future.

Reply:
I've just struggled with the same problem on the "Choose a new super admin" screen, but even more complex as besides those three main paths, there are also some sub-paths within them in regards to back navigation. So I've resolved it by adding a simple history implementation to the machine. Won't go into details now (will explain in the PR for the choose new super admin), just heads up if you're working on some new screens, it may be better to not spend time on `BACK` actions since the proposed implementation won't be compatible with the current way.